bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/models/CollectionSelectComponent.java

```diff
@@ -0,0 +1,51 @@
+package com.adobe.acs.commons.indesign.dynamicdeckdynamo.models;
+
+import com.adobe.acs.commons.indesign.dynamicdeckdynamo.osgiconfigurations.DynamicDeckConfigurationService;
+import com.adobe.acs.commons.indesign.dynamicdeckdynamo.utils.DynamicDeckUtils;
+import com.adobe.acs.commons.mcp.form.SelectComponent;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.scripting.SlingScriptHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * This custom select is used to populate the available collection for the current user.
+ */
+public class CollectionSelectComponent extends SelectComponent {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CollectionSelectComponent.class);
+
+    @Inject
+    @JsonIgnore
+    SlingScriptHelper sling;
+
+    @Override
+    public Map<String, String> getOptions() {
+        Map<String, String> options = new LinkedHashMap<>();
+        ResourceResolver resourceResolver = getHelper().getRequest().getResourceResolver();
+        if (null != resourceResolver) {
+            DynamicDeckConfigurationService configurationService = getHelper().getService(DynamicDeckConfigurationService.class);
```

Comment:
COuld this be gotten using `@OsgiService DynamicDeckConfigurationService configurationService` ... feels weird not using the annotations?

Reply:
That will won't work here. We need to get service from slinghelper

Reply:
Can I ask why?

Reply:
I have given a try but it didn't work for me. Maybe I am missing something.

Actually this class is just to populate the select dropdown and it is called from [DynamicDeckInitioterPageModel.java](https://github.com/Adobe-Consulting-Services/acs-aem-commons/blob/f6cb4b7fd2b01250b2f1256abdc470753f9cc769/bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/models/DynamicDeckInitiatorPageModel.java) which is constructing the form via MCP @formfield from this model. 

`@Inject
    @Named(value = "collectionPath")
    @FormField(
            name = "Collection",
            description = "Select the collection from the drop-down below.",
            component = CollectionSelectComponent.class,
            category = "General")
    private String collectionPath;`
 
As it is not called from any resource directly, it doesn't have 'slingContext'.

Please correct me if I am missing something here and it is possible via other way.

Reply:
My bad. I thought this was a Sling Model, but it's not.